controllers/managedocs_controller.go

```diff
@@ -1752,6 +1762,47 @@ func (r *ManagedOCSReconciler) updateMCGCSV(csv *opv1a1.ClusterServiceVersion) e
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileConsolePlugin() error {
+	r.Log.Info("Reconciling ConsolePlugin")
+	if err := r.get(r.ocpConsolePlugin); err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if r.ocpConsolePlugin.UID == "" {
+		// ODF Operator has to bring this CR,
+		// so if OCP ConsolePlugin is not ready, we ruturn silently
+		r.Log.Info("ConsolePlugin CR not found")
+		return nil
+	}
+	expectedProxy := templates.ConsolePluginProxyTemplate.DeepCopy()
+	expectedProxy.Service.Name = prometheusServiceName
+	expectedProxy.Service.Namespace = r.namespace
+
+	var matchingProxy *consolev1alpha1.ConsolePluginProxy
+	for i := range r.ocpConsolePlugin.Spec.Proxy {
+		currProxy := &r.ocpConsolePlugin.Spec.Proxy[i]
+		if currProxy.Service.Name == expectedProxy.Service.Name &&
+			currProxy.Service.Namespace == expectedProxy.Service.Namespace {
+			matchingProxy = currProxy
+			break
+		}
+	}
+	if matchingProxy == nil {
+		// if we didn't find any matching ConsolePluginProxy, append the expected proxy
+		r.ocpConsolePlugin.Spec.Proxy = append(r.ocpConsolePlugin.Spec.Proxy, *expectedProxy)
+	} else if reflect.DeepEqual(*matchingProxy, *expectedProxy) {
+		// if both, expected and matching PluginProxies, are same
+		// then nothing to do
+		return nil
```

Comment:
This is a strange location for a non-error return. It is not a conditional skip and it is located in the middle of the function. 

This can become problematic if we will introduce, in the future, additional steps at the end of the function that needs to run for all 3 cases. I already suggested a different refactor that solves the problem, in my last review. One that includes an update flag to indicate the need for an update. Is there a reason why you opted out of that suggestion?

Reply:
Made the changes